tools/lib/logreader.py

```diff
@@ -209,6 +209,13 @@ def from_bytes(dat):
     return _LogFileReader("", dat=dat)
 
 
+def get_first(lr: LogIterable, msg_type):
+  for m in lr:
+    if m.which() == msg_type:
+      return getattr(m, msg_type)
+  return None
```

Comment:
```suggestion
def get_first(lr: LogIterable, msg_type):
  return next(filter(lambda m: m.which() == 'carParams', lr), None)
```

can just do this, think there's a few places we already do

Reply:
ah, `CP = next((m.carParams for m in msgs if m.which() == "carParams"), None)` is used in pipeline/lib/fingerprint.py